### PR TITLE
BUG: fix read_gbq lost precision for longs above 2^53 and floats above 10k

### LINKED ISSUE #14020: gbq.py: silently downcasting INTEGER columns to FLOAT is problematic
In many cases INTEGER columns in bigquery are used to store IDs. Such IDs are then likely to be silently corrupted on conversion to dataframe. There seems no good way around this, even if it's clear a priori that the query result will only return non-null INTEGER columns. 

I understand the motivation (the ability to represent NULL values easily and efficiently), but this both seems an undesirable situation (at the very least pandas should give a warning for integers that are not exactly convertible to floats) and inconsistent to how sql queries work in pandas.

I tried to see if it would be possible to consider the 'mode' (NULLABLE or not) of the field when doing the conversion, but that part of the schema information turns out to be of very limited value in bigquery since even trivial projections turn REQUIRED fields into NULLABLE.

I'm not sure what the best solution is, but I think bpq and sql should behave identically with respect to NULLs and that at the very least there should be a warning on information losing integer->float conversions.

So my preference would be to check the returned json query data for nulls and use np.dtype(int64) if there are none in the INTEGER column, and otherwise 'object'. Where that's deemed to inefficient, I think it's better to add a `read_csv` style dtypes argument to allow users to explicitly request float representation for a particular column (or just cast directly in the pseudo-SQL query itself). I'd rather have my code be slower (and possibly see a runtime warning to that effect) than unwittingly corrupt data (without any warning).

### PR DESCRIPTION
**Pull request by tworec**
fixes:
- lost precision for longs above 2^53
- and floats above 10^4

Also contains  `test_gbq.py` clean up
- [x] closes #14020, #14305
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] updated docs
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/gbq.py [hunk 1]
```diff
@@ -553,11 +549,12 @@ def _parse_data(schema, rows):
...
     if field_value is None or field_value == 'null':
         return None
-    if field_type == 'INTEGER' or field_type == 'FLOAT':
+    if field_type == 'INTEGER':
+        return int(field_value)
+    elif field_type == 'FLOAT':
         return float(field_value)
     elif field_type == 'TIMESTAMP':
-        timestamp = datetime.utcfromtimestamp(float(field_value))
-        return np.datetime64(timestamp)
```
Comment (jreback) [inline pandas/io/gbq.py]: why are you changing this?

File: pandas/io/tests/test_gbq.py [hunk 2]
```diff
@@ -430,10 +426,16 @@ def test_should_properly_handle_valid_integers(self):
...
+        query = 'SELECT 4370565731996369294 as VALID_LONG'
+        df = gbq.read_gbq(query, project_id=PROJECT_ID)
+        tm.assert_frame_equal(
+            df, DataFrame({'VALID_LONG': [4370565731996369294]}))
+
     def test_should_properly_handle_null_integers(self):
         query = 'SELECT INTEGER(NULL) as NULL_INTEGER'
         df = gbq.read_gbq(query, project_id=PROJECT_ID)
-        tm.assert_frame_equal(df, DataFrame({'NULL_INTEGER': [np.nan]}))
+        tm.assert_frame_equal(df, DataFrame({'NULL_INTEGER': [None]}))
```
Comment (jreback) [inline pandas/io/tests/test_gbq.py]: this is not correct, ``np.nan`` is the missing entry

**Comment by jreback** [comment]
you are changing lots of things, but have minimal tests for this. pls add some.

**Comment by jreback** [comment]
is this meant to close #14020 ?

File: pandas/io/gbq.py (hunk 1 above)
Comment (tworec) [inline pandas/io/gbq.py]: For two reasons:

1. it is independent from datetime lib
2. it is 4x faster

```
In [1]:
import timeit
timeit.timeit('import numpy as np; from datetime import datetime; np.datetime64(datetime.utcfromtimestamp(float(1234567890.123)))', number=1000000)
Out[1]:
6.163272747769952

In [2]:
timeit.timeit('import numpy as np; np.datetime64(int(float(1234567890.123)*1e6), "us")', number=1000000)
Out[2]:
1.5235848873853683
```

File: pandas/io/tests/test_gbq.py (hunk 2 above)
Comment (tworec) [inline pandas/io/tests/test_gbq.py]: np.NaN can not be stored in np.dtype(int)
http://pandas.pydata.org/pandas-docs/stable/gotchas.html#support-for-integer-na

Comment (jreback) [inline pandas/io/tests/test_gbq.py]: if u have missing values then by definition we cast to float always

we don't use None ever

Comment (jreback) [inline pandas/io/tests/test_gbq.py]: otherwise u end up with an object dtype which is generally not great

**Comment by tworec** [comment]
I didn't see #14020 before, but this PR seems to answer @aschmolck request

File: pandas/io/tests/test_gbq.py (hunk 2 above)
Comment (tworec) [inline pandas/io/tests/test_gbq.py]: casting to float means lost precision for integers > `2^53`.
I think it's better to lost performance than lost precision.
Lost precision means trouble when unnoticed.

I started with solution which raises exception for `NULL` in `INTEGER` column. This forces user to eliminate `NULL`-s (eg. by explicitly casting to `FLOAT` at BigQuery side; there is `float()` function). But finally I think it is less convenient.

My proposed solution is: 
- to store `INTEGER` column without `NULL`-s as `dtype = int64`
- and `INTEGER` with `NULL`-s as `dtype = object`

It mimics behaviour from `read_csv`.  Similarly for `BOOLEAN` columns (bool or object).

File: pandas/io/gbq.py (hunk 1 above)
Comment (jreback) [inline pandas/io/gbq.py]: 1) doesn't really matter, but ok
2) sure, for a single value. you are much better off leaving these as floats, then coercing all in one go if you care about perf (using ``to_datetime``)

File: pandas/io/tests/test_gbq.py [hunk 3]
```diff
@@ -51,10 +51,6 @@ def _skip_if_no_private_key_contents():
         raise nose.SkipTest("Cannot run integration tests without a "
                             "private key json contents")
 
-        _skip_if_no_project_id()
-        _skip_if_no_private_key_path()
```
Comment (jreback) [inline pandas/io/tests/test_gbq.py]: pls restore

File: pandas/io/tests/test_gbq.py (hunk 2 above)
Comment (jreback) [inline pandas/io/tests/test_gbq.py]: this would need to be clearly documented.

**Comment by jreback** [comment]
needs more tests for datetimes

File: pandas/io/gbq.py (hunk 1 above)
Comment (tworec) [inline pandas/io/gbq.py]: It's not the point of this PR. I can restore previous version or leave mine - what do you prefer?

File: pandas/io/tests/test_gbq.py (hunk 3 above)
Comment (tworec) [inline pandas/io/tests/test_gbq.py]: it's a junk code. @parthea already removed these lines in f92cd7e

File: pandas/io/tests/test_gbq.py (hunk 2 above)
Comment (tworec) [inline pandas/io/tests/test_gbq.py]: Ok, so you are ok with this approach?

**Comment by tworec** [comment]
this PR is all about numeric precision loss. I'll restore my datetime improvement. It can be added as separete PR I think.

**Comment by tworec** [comment]
Ok, I've rebased and added full tests for this change (+ test clean up)

**Comment by jorisvandenbossche** [comment]
cc @parthea

**Comment by tworec** [comment]
@jreback, @parthea please review

File: doc/source/io.rst [hunk 4]
```diff
@@ -4428,16 +4428,11 @@ DataFrame with a shape and data types derived from the source table.
 Additionally, DataFrames can be inserted into new BigQuery tables or appended
 to existing tables.
 
-You will need to install some additional dependencies:
-
```
Comment (jreback) [inline doc/source/io.rst]: is this not true anymore?

File: doc/source/io.rst [hunk 5]
```diff
@@ -38,7 +38,7 @@ object.
     * :ref:`read_json<io.json_reader>`
     * :ref:`read_msgpack<io.msgpack>` (experimental)
     * :ref:`read_html<io.read_html>`
-    * :ref:`read_gbq<io.bigquery_reader>` (experimental)
+    * :ref:`read_gbq<io.bigquery>` (experimental)
```
Comment (jreback) [inline doc/source/io.rst]: why did you change this?

File: doc/source/io.rst [hunk 6]
```diff
@@ -4451,22 +4446,58 @@ The key functions are:
...
 
-.. _io.bigquery_reader:
+
+Supported Data Types
+++++++++++++++++++++
+
+Pandas supports these all `BigQuery data types <https://cloud.google.com/bigquery/data-types>`__:
+``STRING``, ``INTEGER`` (64bit), ``FLOAT`` (64 bit), ``BOOLEAN`` and
+``TIMESTAMP`` (microsecond precision). Data types ``BYTES`` and ``RECORD``
+are not supported.
```
Comment (jreback) [inline doc/source/io.rst]: are these unsupported types validated?

File: doc/source/io.rst [hunk 7]
```diff
@@ -4451,22 +4446,58 @@ The key functions are:
...
+
+Integer and boolean ``NA`` handling
++++++++++++++++++++++++++++++++
```
Comment (jreback) [inline doc/source/io.rst]: needs to be the same length as the heading

File: doc/source/io.rst [hunk 8]
```diff
@@ -4451,22 +4446,58 @@ The key functions are:
...
+
+While this trade-off works well for most cases, it breaks down for storing
+values greater than 2**53. Such values in BigQuery can represent identifiers
+and unnoticed precision lost for identifier is what we want to avoid.
+
+Dependencies
+++++++++++++
+
+This module requires these additional dependencies:
+
```
Comment (jreback) [inline doc/source/io.rst]: ahh I c, ok well then make sure the pointer from the installation page goes here.

File: doc/source/io.rst [hunk 9]
```diff
@@ -4451,22 +4446,58 @@ The key functions are:
...
+- `google-api-python-client <http://github.com/google/google-api-python-client>`__
+- `oauth2client <https://github.com/google/oauth2client>`__.
+
 
 .. _io.bigquery_authentication:
 
 Authentication
 ''''''''''''''
 
-.. versionadded:: 0.18.0
```
Comment (jreback) [inline doc/source/io.rst]: don't change things that are not related to this PR.

File: doc/source/io.rst [hunk 10]
```diff
@@ -4475,17 +4506,13 @@ is particularly useful when working on remote servers (eg. jupyter iPython noteb
...
-You will need to install an additional dependency: `oauth2client <https://github.com/google/oauth2client>`__.
-
 Authentication via ``application default credentials`` is also possible. This is only valid
 if the parameter ``private_key`` is not provided. This method also requires that
 the credentials can be fetched from the environment the code is running in.
 Otherwise, the OAuth2 client-side authentication is used.
 Additional information on
 `application default credentials <https://developers.google.com/identity/protocols/application-default-credentials>`__.
 
-.. versionadded:: 0.19.0
```
Comment (jreback) [inline doc/source/io.rst]: why are you taking things out like this? pls don't

File: doc/source/io.rst [hunk 11]
```diff
@@ -4451,22 +4446,58 @@ The key functions are:
...
+
+Since all columns in BigQuery queries are nullable, and NumPy lacks of ``NA``
+support for integer and boolean types, this module will store ``INTEGER`` or
+``BOOLEAN`` columns with at least one ``NULL`` value as ``dtype=object``.
+Otherwise those columns will be stored as ``dtype=int64`` or ``dtype=bool``
+respectively.
+
+This is opposite to default pandas behaviour which will promote integer
+type to float in order to store NAs. See the :ref:`gotchas<gotchas.intna>`
+for detailed explaination.
```
Comment (jreback) [inline doc/source/io.rst]: acutally I disagree with this approach. I would *only* cast to ``object`` if the values are not representable (IOW big), otherwise follow the pandas standard and promot to ``float``. It is much more natural from a user point of view.

File: pandas/io/tests/test_gbq.py [hunk 12]
```diff
@@ -493,6 +497,39 @@ def test_should_read_as_service_account_with_key_contents(self):
                           private_key=_get_private_key_contents())
         tm.assert_frame_equal(df, DataFrame({'VALID_STRING': ['PI']}))
 
+
+class TestReadGBQIntegrationWithServiceAccountKeyPath(tm.TestCase):
```
Comment (jreback) [inline pandas/io/tests/test_gbq.py]: just make a new class in stead of changing existing code.

File: doc/source/whatsnew/v0.19.0.txt [hunk 13]
```diff
@@ -396,8 +396,9 @@ For ``MultiIndex``, values are dropped if any level is missing by default. Speci
 
 Google BigQuery Enhancements
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-The :func:`pandas.io.gbq.read_gbq` method has gained the ``dialect`` argument to allow users to specify whether to use BigQuery's legacy SQL or BigQuery's standard SQL. See the :ref:`docs <io.bigquery_reader>` for more details (:issue:`13615`).
-The :func:`pandas.io.gbq.to_gbq` method now allows the DataFrame column order to differ from the destination table schema (:issue:`11359`).
+- The :func:`pandas.io.gbq.read_gbq` method has gained the ``dialect`` argument to allow users to specify whether to use BigQuery's legacy SQL or BigQuery's standard SQL. See the :ref:`docs <io.bigquery_reader>` for more details (:issue:`13615`).
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.19.0.txt]: Or rather 0.20.0 ? As the current state of this PR changes the behaviour for certain integer columns, so I don't think it is needed to go into 0.19.1

File: doc/source/io.rst (hunk 11 above)
Comment (jreback) [inline doc/source/io.rst]: @tworec we have 96 PR's open at the moment. I don't always comment till someone indicates things are ready.

can you *see* whether the column is nullable or not a-priori?

File: doc/source/whatsnew/v0.19.0.txt (hunk 13 above)
Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: @jorisvandenbossche makes a good point, so 0.20.0

File: doc/source/io.rst (hunk 4 above)
Comment (jreback) [inline doc/source/io.rst]: then you need a pointer from the install.rst to the deps section

**Comment by jreback** [comment]
I think it might be better to force the user to specify a ``dtype`` keyword to override / set behavior for particular columns (like we do in ``read_csv``; we sort of do this with ``coerce_float`` in ``read_sql``.

@jorisvandenbossche

File: doc/source/io.rst (hunk 11 above)
Comment (tworec) [inline doc/source/io.rst]: ok, I understand :)

no, all columns in BigQuery queries are nullable

File: doc/source/whatsnew/v0.19.0.txt (hunk 13 above)
Comment (tworec) [inline doc/source/whatsnew/v0.19.0.txt]: ok

**Comment by tworec** [comment]
I offer solution with default behavior. 
Specifing `dtype` is cool, but it's new feature which I do not want to focus on right now. This PR is ment to fix things up, not to put new features.

Can we stop here?

**Comment by tworec** [comment]
@jreback ping

File: doc/source/io.rst [hunk 14]
```diff
@@ -4452,7 +4447,43 @@ The key functions are:
...
+and unnoticed precision lost for identifier is what we want to avoid.
+
+Dependencies
+++++++++++++
+
+This module requires these additional dependencies:
+
+- `httplib2 <http://pypi.python.org/pypi/httplib2>`__
+- `google-api-python-client <http://github.com/google/google-api-python-client>`__
+- `oauth2client <https://github.com/google/oauth2client>`__.
```
Comment (jreback) [inline doc/source/io.rst]: is ``oauth2client`` optional?

File: doc/source/io.rst (hunk 4 above)
Comment (jreback) [inline doc/source/io.rst]: you need to add a pointer from here to the Dependency section of the docs (that you added below).

File: doc/source/io.rst (hunk 14 above)
Comment (tworec) [inline doc/source/io.rst]: no, it is required.

File: doc/source/whatsnew/v0.19.1.txt [hunk 15]
```diff
@@ -0,0 +1,87 @@
+.. _whatsnew_0191:
+
```
Comment (tworec) [inline doc/source/whatsnew/v0.19.1.txt]: sure

File: doc/source/whatsnew/v0.19.1.txt [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+
+
+
+.. _whatsnew_0191.bug_fixes:
+
+Bug Fixes
+~~~~~~~~~
+
+- The :func:`pandas.io.gbq.to_gbq` method now stores ``INTEGER`` columns as ``dtype=object`` if they contain ``NULL`` values. Othewise they are stored as ``int64``. This prevents precision lost for integers greather than 2**53. Furthermore ``FLOAT`` columns with values above 10**4 are no more casted to ``int64`` which also caused precision lost (:issue: `14064`).
```
Comment (tworec) [inline doc/source/whatsnew/v0.19.1.txt]: ok, I'm moving it to 0.20.0

**Comment by tworec** [comment]
@jreback ok, I've rebased, moved to 0.20.0 and added install page ref to gbq deps

File: doc/source/io.rst [hunk 17]
```diff
@@ -4452,7 +4447,44 @@ The key functions are:
 
 .. currentmodule:: pandas
 
-.. _io.bigquery_reader:
+
+Supported Data Types
+++++++++++++++++++++
+
+Pandas supports these all `BigQuery data types <https://cloud.google.com/bigquery/data-types>`__:
```
Comment (jorisvandenbossche) [inline doc/source/io.rst]: "these all" -> "all these" ?

File: doc/source/io.rst [hunk 18]
```diff
@@ -4452,7 +4447,44 @@ The key functions are:
...
+``STRING``, ``INTEGER`` (64bit), ``FLOAT`` (64 bit), ``BOOLEAN`` and
+``TIMESTAMP`` (microsecond precision). Data types ``BYTES`` and ``RECORD``
+are not supported.
+
+Integer and boolean ``NA`` handling
++++++++++++++++++++++++++++++++++++
+
+.. versionadded:: 0.19
```
Comment (jorisvandenbossche) [inline doc/source/io.rst]: 0.20

File: doc/source/whatsnew/v0.20.0.txt [hunk 19]
```diff
@@ -80,3 +80,5 @@ Performance Improvements
 
 Bug Fixes
 ~~~~~~~~~
+
+- The :func:`pandas.io.gbq.to_gbq` method now stores ``INTEGER`` columns as ``dtype=object`` if they contain ``NULL`` values. Othewise they are stored as ``int64``. This prevents precision lost for integers greather than 2**53. Furthermore ``FLOAT`` columns with values above 10**4 are no more casted to ``int64`` which also caused precision lost (:issue: `14064`).
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.20.0.txt]: `pandas.io.gbq.to_gbq` -> `~DataFrame.to_gbq`

**Comment by jorisvandenbossche** [comment]
@tworec Added some minor doc comments. 
But, we also still have to decide about object vs float (https://github.com/pandas-dev/pandas/pull/14064#discussion_r82159392), I think? I find it a bit strange that we would take here a different approach than in all other places in pandas.

File: doc/source/io.rst (hunk 17 above)
Comment (tworec) [inline doc/source/io.rst]: :) sorry, it's my polish 2 english calque

File: doc/source/whatsnew/v0.20.0.txt (hunk 19 above)
Comment (tworec) [inline doc/source/whatsnew/v0.20.0.txt]: ok

File: doc/source/io.rst (hunk 18 above)
Comment (tworec) [inline doc/source/io.rst]: ok

**Comment by tworec** [comment]
@jorisvandenbossche I've corrected docs.

regarding float storage: my proposition is described in comment and implemented with tests
whats your proposition?

**Comment by jreback** [comment]
@tworec sorry we got a bit lost in the shuffle.

can you rebase and we can get this in.

**Comment by tworec** [comment]
@jreback ok, I've rebased

**Comment by jreback** [comment]
@parthea thoughts?

**Comment by tworec** [comment]
hi @jreback , can we finally merge this? Please decide if you want this change or drop it.

**Comment by jreback** [comment]
I was waiting for @parthea to come back on this.

**Comment by tworec** [comment]
oh, I c. We'll wait then. 
@parthea can you please check this out? :)

File: pandas/io/tests/test_gbq.py [hunk 20]
```diff
@@ -517,18 +554,72 @@ def test_should_properly_handle_valid_integers(self):
...
+        df = gbq.read_gbq(query, project_id=_get_project_id(),
+                          private_key=_get_private_key_path())
+        tm.assert_frame_equal(df, DataFrame(
+            {'VALID_DOUBLE': [pi * 10 ** 307]}))
+
+    def test_should_properly_handle_nullable_doubles(self):
+        from math import pi
+        query = '''select * from
+                    (select PI() * POW(10, 307) as NULLABLE_DOUBLE),
+                    (select null as NULLABLE_DOUBLE)'''
```
Comment (parthea) [inline pandas/io/tests/test_gbq.py]: SQL reserved words should be upper case for consistency with existing code.  We should either use the following or change existing code if there is a preference to change the SQL formatting.
```
        query = '''SELECT * FROM
                    (SELECT PI() * POW(10, 307) AS Nullable_Double),
                    (SELECT NULL AS Nullable_Double)'''
```

File: pandas/io/tests/test_gbq.py [hunk 21]
```diff
@@ -517,18 +554,72 @@ def test_should_properly_handle_valid_integers(self):
...
                           private_key=_get_private_key_path())
         tm.assert_frame_equal(df, DataFrame(
-            {'VALID_FLOAT': [3.141592653589793]}))
+            {'VALID_FLOAT': [pi]}))
+
+    def test_should_properly_handle_nullable_floats(self):
+        from math import pi
+        query = '''select * from
+                    (select PI() as NULLABLE_FLOAT),
+                    (select null as NULLABLE_FLOAT)'''
```
Comment (parthea) [inline pandas/io/tests/test_gbq.py]: SQL keywords should be uppercase for consistency with existing code.

File: pandas/io/tests/test_gbq.py [hunk 22]
```diff
@@ -1124,7 +1228,7 @@ def setUpClass(cls):
         # executing *ALL* tests described below.
 
         _skip_if_no_project_id()
-        _skip_if_no_private_key_path()
+        _skip_local_auth_if_in_travis_env()
 
         _setup_common()
         clean_gbq_environment(_get_private_key_path())
```
Comment (parthea) [inline pandas/io/tests/test_gbq.py]: Since we are testing local user account auth, can we change this to `clean_gbq_environment()`? We should also make the same change in tearDownClass.

**Review by parthea — COMMENTED** [review]
I added a few minor comments. I think `_get_private_key_path()` should not be called in `TestToGBQIntegrationWithLocalUserAccountAuth`. 

Separately, it would be great if you could follow the steps in the [contributing docs to run the integration tests from your fork on Travis-CI](http://pandas.pydata.org/pandas-docs/stable/contributing.html#running-google-bigquery-integration-tests). You only need to do the Travis configuration described in the contributing docs once and you're all set for future PRs.

**Review by tworec — COMMENTED** [review]
I fixed things as you @parthea suggested. Please review once again.

I don't fully trust travis CI so I need to create separete empty project for this purpose. And I'm not on my own. My company need to set up billing for it. It will take time.

**Comment by tworec** [comment]
@parthea please review, because every time I look at this change it need manual rebase (conflicts).

**Comment by tworec** [comment]
@parthea my Travis CI setup is broken, it causes errors with pickles eg.
`ERROR: pandas.io.tests.test_pickle.TestPickle.test_pickles('0.14.0',)`

https://travis-ci.org/RTBHOUSE/pandas/jobs/199354963

Can you help?

**Comment by parthea** [comment]
@tworec I'm sorry for the delay. The latest commit looks good. 

Could you try running `git push --tags` on your master branch to see if that resolves the issue ?  Let me know if this works. I will submit a PR to update the BigQuery integration testing steps in the contributing docs to make it clear that you need to push tags to your master branch.

**Comment by tworec** [comment]
@jreback thx

I've done it. Output is below. Why is this needed -- tags should not affect building right?
I guess this 141 commits is/was the problem. We'll see.

```
$ g co master 
Switched to branch 'master'
Your branch is ahead of 'origin/master' by 141 commits.
  (use "git push" to publish your local commits)

$ g ps --tags
To git@github.com:RTBHOUSE/pandas.git
 * [new tag]         v0.16.2 -> v0.16.2
 * [new tag]         v0.17.0 -> v0.17.0
 * [new tag]         v0.17.0rc1 -> v0.17.0rc1
 * [new tag]         v0.17.0rc2 -> v0.17.0rc2
 * [new tag]         v0.17.1 -> v0.17.1
 * [new tag]         v0.18.0 -> v0.18.0
 * [new tag]         v0.18.0rc1 -> v0.18.0rc1
 * [new tag]         v0.18.0rc2 -> v0.18.0rc2
 * [new tag]         v0.18.1 -> v0.18.1
 * [new tag]         v0.19.0 -> v0.19.0
 * [new tag]         v0.19.0rc1 -> v0.19.0rc1
 * [new tag]         v0.19.1 -> v0.19.1
 * [new tag]         v0.19.2 -> v0.19.2

```

**Comment by tworec** [comment]
i've squashed

**Comment by jreback** [comment]
@tworec when git clones a repo it doesn't clone the tags, so they are from *your* master whenever it was cloned, NOT the current one. This behavior baffles me, but that's how it works :<

**Comment by tworec** [comment]
ok, now my travis build works well
thx for explanation

**Comment by jreback** [comment]
thanks for the patience @tworec 

you can also put up a PR to: https://github.com/pydata/pandas-gbq (will need to take out the documentation changes, but outherwise should be clean).
